Mathlib/Algebra/Group/Submonoid/Basic.lean

```diff
@@ -408,16 +408,15 @@ and verify that `p x` and `p y` imply `p (x * y)`. -/
       `x ∈ s`, verify `p 0`, and verify that `p x` and `p y` imply `p (x + y)`."]
 theorem dense_induction {p : M → Prop} (x : M) {s : Set M} (hs : closure s = ⊤)
```

Comment:
Is this the only "weird" induction principle? If so, I think an `example` after it showing how to use it would be a good idea.

Reply:
Should `s` be explicit, do you think? This would allow the user to defer the proof of `closure s = ⊤` into the arms of the induction without creating a metavariable.

Reply:
What would the `example` look like if this were explicit?

Reply:
`induction x using dense_induction s`

Reply:
I've made `s` explicit and added an example.